### DISCUSSION TITLE
Updating the monorepo to node v20

### DISCUSSION THREAD

**Discussion by rrennick** [woocommerce/woocommerce#45529] [Request for Comments]
We are getting deprecated notices in our CI runs:

<img width="698" alt="Screenshot 2024-03-12 at 3 22 40 PM" src="https://github.com/woocommerce/woocommerce/assets/343847/921dc4b3-6200-41ac-a229-5c285356521b">

Github plans to [start enforcing node v20](https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/) on CI workflows on May 13, 2024.

I created a [POC PR](https://github.com/woocommerce/woocommerce/pull/45148) that you can use to test node v20 against your daily workflow/activities. This was rebased to trunk as of March 11, 2024. I will rebase the branch weekly.

We would like to update node to v20 in early April. Please leave a comment here if you encounter any issues during testing.

**Comment by rrennick** [comment]
Noting that #43034 needs to be addressed before node can be updated.

**Comment by lsinger** [comment]
Heads up that I did some testing for the workflows used by Proton[^1] and didn't encounter any issues under node v20. ✅ 

[^1]: (`pnpm install`; `pnpm build`; `pnpm env:test`; `pnpm test:php:env`; making changes to (legacy) SCSS using `pnpm watch:build`)

**Comment by louwie17** [comment]
I did similar tests to @lsinger, and also didn't encounter any issues.

**Comment by adrianduffell** [comment]
Nice, thanks @rrennick. It tested well for me too. I tried a few other commands including `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example`; `pnpm --filter=@woocommerce/storybook run storybook`

**Comment by nima-karimi** [comment]
Thanks @rrennick. I've tested @woocommerce/singularity's plugin and packages and didn't notice any issues.

**Comment by danielwrobert** [comment]
Thanks for working on this @rrennick!

I've tested the workflows for Woo Blocks (Origami & Kirigami) and did not encounter any issues.

I ran the following commands:

- `pnpm install`
- `pnpm --filter=@woocommerce/block-library watch:build`
- `pnpm --filter=@woocommerce/plugin-woocommerce watch:build`
- `pnpm --filter=@woocommerce/block-library build`
- `pnpm --filter='@woocommerce/plugin-woocommerce' build`

It looks like @samueljseay also tested these workflows and noted in https://github.com/woocommerce/woocommerce/pull/45148#issuecomment-1996752510.
